File: 59-spiral-matrix-ii/spiral-matrix-ii.java
class Solution {
    public int[][] generateMatrix(int n) {

        int colStart = 0;
        int colEnd = n-1;
        int rowStart = 0;
        int rowEnd = n-1;
        int count = 1;
        int matrix[][] = new int[n][n];

    while(count<=n*n){

        //right
        for(int i=colStart;i<=colEnd;i++){
            matrix[rowStart][i] = count;
            count++;
        }
        rowStart++;

        //down
        for(int i=rowStart;i<=rowEnd;i++){
            matrix[i][colEnd] = count;
            count++;
        }
        colEnd--;

        //left
        for(int i=colEnd;i>=colStart;i--){
            matrix[rowEnd][i] = count;
            count++;
        }
        rowEnd--;

        //up
        for(int i=rowEnd;i>=rowStart;i--){
            matrix[i][colStart] = count;
            count++;
        }
        colStart++;
    }

    return matrix;
        
    }
}

File: 59-spiral-matrix-ii/spiral-matrix-ii.ts
function generateMatrix(n: number): number[][] {

    let colStart = 0;
    let colEnd = n-1;
    let rowStart = 0;
    let rowEnd = n-1;
    let count = 1;
    let matrix = Array.from({length: n}, ()=>new Array(n))

    while(count<=n*n){

        //right
        for(let i=colStart;i<=colEnd;i++){
            matrix[rowStart][i] = count;
            count++;
        }
        rowStart++;

        //down
        for(let i=rowStart;i<=rowEnd;i++){
            matrix[i][colEnd] = count;
            count++;
        }
        colEnd--;

        //left
        for(let i=colEnd;i>=colStart;i--){
            matrix[rowEnd][i] = count;
            count++;
        }
        rowEnd--;

        //up
        for(let i=rowEnd;i>=rowStart;i--){
            matrix[i][colStart] = count;
            count++;
        }
        colStart++;
    }

    return matrix
    
};